### PR TITLE
chore: optional start some network service

### PR DESCRIPTION
**Pull request by quake**
optional start `OutboundPeerService` and `DnsSeedingService`, and fix random failure integration test `BlockSyncDuplicatedAndReconnect `

### CODE REVIEW

**Comment by quake** [comment]
@nervos-bot-user try integration

File: network/src/network.rs [hunk 1]
```diff
@@ -815,21 +815,24 @@ impl NetworkService {
...
-            network_state.config.dns_seeds.clone(),
-        );
-        let bg_services = vec![
+        let mut bg_services = vec![
             Box::new(ping_service.for_each(|_| Ok(()))) as Box<_>,
             Box::new(disc_service) as Box<_>,
-            Box::new(outbound_peer_service) as Box<_>,
-            Box::new(dns_seeding_service) as Box<_>,
         ];
+        if config.connect_outbound_interval_secs > 0 {
```
Comment (keroro520) [inline network/src/network.rs]: We should add documentation or some comments about `connect_outbound_interval_secs == 0`

Comment (keroro520) [reply network/src/network.rs]: Or wrap a function `config.enable_auto_connect_outbound() -> bool;`

Comment (quake) [reply network/src/network.rs]: Updated, please review again, thanks.

**Pull request merged**